### PR TITLE
`data_to_wide`: drop cols that are not in `id_cols` (if specified), `names_from` or `values_from`

### LINKED ISSUE #293: `data_to_wide` fails with example
No problem with current CRAN version (0.6.2) but fails with dev version.

``` r
suppressPackageStartupMessages({
  library(tidyr)
  library(dplyr)
})

weekdays <- c("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")

daily <- tibble(
  day = factor(c("Tue", "Thu", "Fri", "Mon"), levels = weekdays),
  value = c(2, 3, 1, 5)
)

daily
#> # A tibble: 4 × 2
#>   day   value
#>   <fct> <dbl>
#> 1 Tue       2
#> 2 Thu       3
#> 3 Fri       1
#> 4 Mon       5

pivot_wider(daily, names_from = day, values_from = value)
#> # A tibble: 1 × 4
#>     Tue   Thu   Fri   Mon
#>   <dbl> <dbl> <dbl> <dbl>
#> 1     2     3     1     5

datawizard::data_to_wide(daily, names_from = "day", values_from = "value")
#> # A tibble: 1 × 4
#>     Tue   Thu   Fri   Mon
#>   <dbl> <dbl> <dbl> <dbl>
#> 1     1     5     3     2
```

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #293

### CODE REVIEW

File: R/data_to_wide.R [hunk 1]
```diff
@@ -152,7 +156,7 @@ data_to_wide <- function(data,
   n_rows_per_group <- table(new_data$temporary_id)
   n_values_per_group <- insight::n_unique(n_rows_per_group)
 
-  not_all_cols_are_selected <- length(not_selected) > 0
+  not_all_cols_are_selected <- length(id_cols) > 0
```
Comment (IndrajeetPatil) [inline R/data_to_wide.R:159]:
```suggestion
  not_all_cols_are_selected <- length(id_cols) > 0L
```

File: R/data_to_wide.R [hunk 2]
```diff
@@ -138,10 +142,10 @@ data_to_wide <- function(data,
 
   # create an id with all variables that are not in names_from or values_from
   # so that we can create missing combinations between this id and names_from
-  if (length(not_selected) > 1) {
-    new_data$temporary_id <- do.call(paste, c(new_data[, not_selected, drop = FALSE], sep = "_"))
-  } else if (length(not_selected) == 1) {
-    new_data$temporary_id <- new_data[[not_selected]]
+  if (length(id_cols) > 1) {
```
Comment (IndrajeetPatil) [inline R/data_to_wide.R:145]:
```suggestion
  if (length(id_cols) > 1L) {
```

**Review by IndrajeetPatil — APPROVED** [review]
Thanks!

Few nits to resolve, but otherwise looks good.

**Comment by etiennebacher** [comment]
Thank you, why is it necessary to change from `1` to `1L`? I know that `1L` means integers and not just numbers but I've never found a case where it makes a difference

**Comment by IndrajeetPatil** [comment]
It doesn't make a difference but it does produce a lint: https://lintr.r-lib.org/dev/reference/implicit_integer_linter.html

This is desirable for two reasons:

- you are saving some computations by avoiding R having to do implicit conversions when needed (efficiency)

- it makes it clear that the quantities in question can never exist as fractions (safety)

**Pull request merged**